javascript/react_js/advanced_concepts.md

```diff
@@ -40,15 +40,11 @@ You might have already heard about Redux. [Redux](https://react-redux.js.org) is
 
 As your application becomes larger and you have multiple components to ensure reusability, you might find yourself passing props through a lot of components in the middle or you might have a lot of components that require the same props. This pattern is known as `prop drilling`. To help avoid this, React provides the Context API that lets a parent component provide data to all the components in its tree without having to pass props. Imagine that you have implemented an optional dark theme in your website and quite a few of your components require this theme data to correctly style itself - having a `context` of your theme will allow all children components to have access to this data.
 
-The [Passing Data Deeply with Context webpage](https://beta.reactjs.org/learn/passing-data-deeply-with-context) on React Docs does an excellent job of explaining on how to use the Context API. Furthermore, if you wish to learn more advanced concepts in React, [React Docs](https://react.dev/) is a great resource to dig into.
+The [Passing Data Deeply with Context webpage](https://react.dev/learn/passing-data-deeply-with-context) on React Docs does an excellent job of explaining on how to use the Context API. Furthermore, if you wish to learn more advanced concepts in React, [React Docs](https://react.dev/) is a great resource to dig into.
 
-#### <span id="higher-order-components">5. Higher-order Components</span>
+#### <span id="more-hooks">5. More Hooks</span>
 
-[Higher-order components](https://reactjs.org/docs/higher-order-components.html) are components that consume another component and return a third component. This [article on HOC's by Smashing Magazine](https://www.smashingmagazine.com/2020/06/higher-order-components-react/) shows different use cases and provides some great comparisons to help you understand.
-
-#### <span id="more-hooks">6. More Hooks</span>
-
-There are a lot of [hooks](https://reactjs.org/docs/hooks-reference.html) out there, with the number only increasing by the day (since you can write your own!). As you have seen in previous lessons, it seems like the React team wants us to use more functional components with hooks in the future, so it is definitely worth getting to know both the hooks that are built-in to React, as well as how to create your [own](https://reactjs.org/docs/hooks-custom.html).
+There are a lot of [hooks](https://react.dev/reference/react) out there, with the number only increasing by the day (since you can write your own!). As you have seen in previous lessons, it seems like the React team wants us to use more functional components with hooks in the future, so it is definitely worth getting to know both the hooks that are built-in to React, as well as how to create your [own](https://react.dev/learn/reusing-logic-with-custom-hooks).
```

Comment:
This part is fine. I did notice that higher-order components is mentioned in the learning outcomes. If you could go ahead and remove that as well that would be awsome.

Reply:
I have removed the higher-order components from learning outcomes also in the knowledge check section at the end of the guide, it references Higher-Order-Component section, removed that too. Please review it and guide me if i need to do some other changes.